applications/nrf_desktop/CMakeLists.txt

```diff
@@ -6,27 +6,23 @@
 
 cmake_minimum_required(VERSION 3.8.2)
 
-set(NRF_SUPPORTED_BOARDS
-  nrf52810_pca20045
-  nrf52840_pca20041
-  nrf52840_pca10056
-  nrf52840_pca10059
-  nrf52_pca20037
-  nrf52_pca20044
-  )
-
-set(NRF_SUPPORTED_BUILD_TYPES
-  ZDebug
-  ZDebugWithShell
-  ZDebugMCUBoot
-  ZRelease
-  ZReleaseMCUBoot
-  )
-
 if (NOT CMAKE_BUILD_TYPE)
   set(CMAKE_BUILD_TYPE ZDebug)
 endif()
 
+if (NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/configuration/${BOARD}")
```

Comment:
@pdunaj Why wasn't I invited here ?
This solution doesn't consider the fact that users could be using boards that uses the revision feature.

Reply:
Hi @tejlmand , I don't know to be honest. That was around time we were helping FAEs to learn the thing and have first products go. Probably overlooked.

I expect we have not anticipated board revision.

Reply:
ok, let's look into this after next upmerge.

With next upmerge we'll have Zephyr CMake modules infrastructure, and I believe that we could benefit from that for this feature and the configuration/board scheme.